### ISSUE TITLE
foreign key properties

### ISSUE THREAD

**Issue by dimitri-yatsenko** [datajoint/datajoint-matlab#74]
I propose to allow specification of foreign key properties `unique` and `required`.

I propose the following syntax.

```
-> [unique, optional] lab.Animal 
```

"Unique" means that each animal in `lab.Animal` can be referenced only once from the referencing table.

"Optional" means that the foreign key values can be null.  When null, the unique constraint is not applicable. 

By default, foreign keys are not unique and not optional. 

This syntax is compatible with the syntax for renamed foreign keys:

```
(subject_id) -> [unique] lab.Animal
```

**Comment by ixcat** [comment]
superseded by https://github.com/datajoint/datajoint-matlab/issues/110 which has more up to date spec from existing DJ python
